lib/LanguageServerCodeTransform/LspCommand/ExtractMethodCommand.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerCodeTransform\LspCommand;
+
+use Amp\Promise;
+use Amp\Success;
+use Phpactor\CodeTransform\Domain\Exception\TransformException;
+use Phpactor\CodeTransform\Domain\Refactor\ExtractMethod;
+use Phpactor\CodeTransform\Domain\SourceCode;
+use Phpactor\Extension\LanguageServerBridge\Converter\TextEditConverter;
+use Phpactor\LanguageServerProtocol\ApplyWorkspaceEditResponse;
+use Phpactor\LanguageServer\Core\Command\Command;
+use Phpactor\LanguageServerProtocol\WorkspaceEdit;
+use Phpactor\LanguageServer\Core\Server\ClientApi;
+use Phpactor\LanguageServer\Core\Workspace\Workspace;
+
+class ExtractMethodCommand implements Command
+{
+    public const NAME  = 'extract_method';
+    public const DEFAULT_METHOD_NAME  = 'newMethod';
+
+    /**
+     * @var ExtractMethod
+     */
+    private $extractMethod;
+    /**
+     * @var ClientApi
+     */
+    private $clientApi;
+    /**
+     * @var Workspace
+     */
+    private $workspace;
+
+    public function __construct(
+        ClientApi $clientApi,
+        Workspace $workspace,
+        ExtractMethod $extractMethod
+    ) {
+        $this->extractMethod = $extractMethod;
+        $this->clientApi = $clientApi;
+        $this->workspace = $workspace;
+    }
+    /**
+     * @return Promise<ApplyWorkspaceEditResponse>
+     */
+    public function __invoke(string $uri, int $startOffset, int $endOffset): Promise
+    {
+        $textDocument = $this->workspace->get($uri);
+        
+        try {
+            $textEdits = $this->extractMethod->extractMethod(
+                SourceCode::fromStringAndPath($textDocument->text, $textDocument->uri),
+                $startOffset,
+                $endOffset,
+                self::DEFAULT_METHOD_NAME
```

Comment:
it seems we canot currently ask the user [for input](https://github.com/microsoft/language-server-protocol/issues/1164). Assuming this method will throw an exception if there is an existing method, you could add an increment to the default method name.

Reply:
Actually my guess was, since `myMethod` is so silly of a name, if you have one of these lying around, you have some more work to do until you extract another one. Go rename the other one, _then_ extract this one :-) Kind of a code quality error. Otherwise doing a counter would be trivial, I think, if "method exists" is a catchable exception.

The perfect solution would be to either ask for input (I saw you found the same discussion), or enter rename mode immediately after completing. Probably it's best to wait for the former.

So, code quality wise, I'd leave it as it is, but I am happy to do a counter if think it'll be better.